src/Forward.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef IODRIVERS_BASE_FORWARD_HPP
+#define IODRIVERS_BASE_FORWARD_HPP
+
+#include <base/Time.hpp>
+
+namespace iodrivers_base {
+    class Driver;
+
+    /** Forward data between two subclasses of iodrivers_base::Driver
+     *
+     * It blocks until one of the drivers is closed
+     *
+     * @param raw_mode whether bytes are read using readRaw (true) or readPacket(false)
+     * @param driver1 one of the two drivers
+     * @param timeout1 how long we should wait, while reading, before
```

Comment:
Fragmentation (if any) will be inherent to the transport being used. The user will have to deal with it regardless (assuming he/she won't be expecting something joining packets in the middle). So, I would rather just ignore the potential fragmentation then introduce a delay between sends.

Also, for the raw mode thing, if the user is using UDP, I think it is reasonable for him/her to expect the boundaries to be respected (i.e no joined datagrams)

Reply:
> Fragmentation (if any) will be inherent to the transport being used.

Confusing wording. By "fragmentation", I meant "avoid sending stuff byte-by-byte" when reading from e.g. serial lines, which would have a terrible performance on any network.

A 1ms "delay" is enough. It's basically corking, but adapted to the exact setup

Reply:
Good point. Also, one can always pass a zero timeout to avoid the delay (or the joined datagrams)...